ddsreplayer/test/blackbox/mcap/McapFileReadTest.cpp

```diff
@@ -0,0 +1,178 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// #define MCAP_IMPLEMENTATION
+
+#include <cpp_utils/testing/gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <cpp_utils/event/MultipleEventHandler.hpp>
+#include <cpp_utils/logging/CustomStdLogConsumer.hpp>
+
+#include "dds/HelloWorldSubscriber.h"
+
+#include "tool/DdsReplayer.hpp"
+
+#include <iostream>
+#include <thread>
+#include <chrono>
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastrtps::rtps;
+using namespace eprosima::fastrtps;
+using namespace eprosima::ddsrecorder::replayer;
+
+namespace test {
+
+// Publisher
+
+const unsigned int DOMAIN = 110;
+
+std::string topic_name = "/dds/topic";
+
+} // test
+
+
+void create_subscriber_replayer(
+        DataToCheck& data,
+        std::string configuration_path = "resources/config_file.yaml")
+{
+
+    // Create replayer instance
+    std::string input_file = "resources/helloworld_file.mcap";
+    {
+        // Create Subscriber
+        HelloWorldSubscriber subscriber(
```

Comment:
Write a detailed comment explaining why this order of construction and destruction.